modules/programs/neovim.nix

```diff
@@ -287,10 +318,22 @@ in {
         configure.packages.*.opt  -> programs.neovim.plugins = [ { plugin = ...; optional = true; }]
         configure.packages.*.start  -> programs.neovim.plugins = [ { plugin = ...; }]
         configure.customRC -> programs.neovim.extraConfig
-    '';
+    ''
+    ++ optional (filter (p: isDerivation p) cfg.plugins != []) ''
+      All plugins should now be of type 'pluginWithConfigType' and not a package anymore, e.g.,
+        plugins = [ plenary-nvim ];
+      should now be:
+        plugins = [ { plugin = plenary-nvim; } ];
+    ''
+    ;
```

Comment:
See my comment about the new option `plugins.*.type`.

Reply:
I added the warning because I manually change "plenary-nvim" in a pluginWithConfigType type and it's tricky.
There must be some function in nixpkgs that can do that which would be more reliable such that we can keep both. Suggestions welcome, else I'll look when I have time.

Reply:
I've removed the deprecation for now as it is not necessary